dnf/package.py

```diff
@@ -225,9 +225,9 @@ def schemes_filter(url_list):
                 if schemes:
                     s = dnf.pycomp.urlparse.urlparse(url)[0]
                     if s in schemes:
-                        return url + self.location
+                        return os.path.join(url, self.location)
```

Comment:
I think correct fix here is to use urljoin()... I will merge other part of this PR now and will leave for you to fix this one later